File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeEventHandler.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.chat;

/**
 * {@link ChatCompositeEventHandler}&lt;T&gt;
 *
 * <p>A generic handler for call composite events.</p>
 * <p> - {@link ChatAdapter#addOnErrorEventHandler(ChatCompositeEventHandler)} for Error Handling</p>
 * for Remote Participant Join Notifications</p>
 *
 * @param <T> The callback event Type.
 */
public interface ChatCompositeEventHandler<T> {
    /**
     * A callback method to process error event of type T
     *
     * @param eventArgs {@link T}
     */
    void handle(T eventArgs);
}

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeEventHandler.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package com.azure.android.communication.ui.chat

/**
 * [ChatCompositeEventHandler]&lt;T&gt;
 *
 *
 * A generic handler for chat composite events.
 *
 * @param <T> The callback event Type.
</T> */
internal interface ChatCompositeEventHandler<T> {
    /**
     * A callback method to process error event of type T
     *
     * @param eventArgs [T]
     */
    fun handle(eventArgs: T)
}
